core/server/master/src/test/java/alluxio/master/metastore/BlockStoreTest.java

```diff
@@ -11,37 +11,103 @@
 import alluxio.master.metastore.rocks.RocksBlockStore;
 import alluxio.proto.meta.Block;
 import alluxio.resource.CloseableIterator;
 
-import com.google.common.io.Files;
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.Parameterized;
+import org.rocksdb.RocksDBException;
 
+import java.io.File;
+import java.nio.charset.Charset;
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.HashMap;
 import java.util.List;
+import java.util.function.Supplier;
 
 @RunWith(Parameterized.class)
 public class BlockStoreTest {
+  private static final String CONF_NAME = "/rocks-block.ini";
+  private static String sDir;
+
   @Parameterized.Parameters
-  public static Collection<Object[]> data() throws Exception {
-    return Arrays.asList(new Object[][] {
-        {new RocksBlockStore(Files.createTempDir().getAbsolutePath())},
-        {new HeapBlockStore()}
-    });
+  public static Collection<Supplier<BlockStore>> data() throws Exception {
+    sDir = AlluxioTestDirectory.createTemporaryDirectory(
+        "block-store-test").getAbsolutePath();
+    File confFile = new File(sDir + CONF_NAME);
+    writeStringToFile(confFile, ROCKS_CONFIG, (Charset) null);
+    return Arrays.asList(
+        () -> new RocksBlockStore(sDir),
+        HeapBlockStore::new
+    );
   }
 
   @Parameterized.Parameter
+  public Supplier<BlockStore> mBlockStoreSupplier;
   public BlockStore mBlockStore;
 
+  @Before
+  public void before() {
+    mBlockStore = mBlockStoreSupplier.get();
+  }
+
+  @After
+  public void after() {
+    mBlockStore.close();
+  }
+
+  @Test
+  public void rocksConfigFile() throws Exception {
+    assumeTrue(mBlockStore instanceof RocksBlockStore);
+    // close the store first because we want to reopen it with the new config
+    mBlockStore.close();
+    try (AutoCloseable ignored = new ConfigurationRule(new HashMap<PropertyKey, Object>() {
+      {
+        put(PropertyKey.ROCKS_BLOCK_CONF_FILE, sDir + CONF_NAME);
+      }
+    }, ServerConfiguration.global()).toResource()) {
+      mBlockStore = mBlockStoreSupplier.get();
+      testPutGet();
+    }
+  }
+
+  @Test
+  public void rocksInvalidConfigFile() throws Exception {
+    assumeTrue(mBlockStore instanceof RocksBlockStore);
+    // close the store first because we want to reopen it with the new config
+    mBlockStore.close();
+    // write an invalid config
+    String path = sDir + CONF_NAME + "invalid";
+    File confFile = new File(path);
+    writeStringToFile(confFile, "Invalid config", (Charset) null);
+
+    try (AutoCloseable ignored = new ConfigurationRule(new HashMap<PropertyKey, Object>() {
+      {
+        put(PropertyKey.ROCKS_BLOCK_CONF_FILE, path);
+      }
+    }, ServerConfiguration.global()).toResource()) {
+      assertEquals(RocksDBException.class,
+          assertThrows(RuntimeException.class, this::before).getCause().getClass());
```

Comment:
```suggestion
      RuntimeException exception = assertThrows(RuntimeException.class, this::before);
      assertEquals(RocksDBException.class, exception.getCause().getClass());
```
I think splitting it up into two `assert` checks makes the test more legible.